### PR TITLE
apps/examples/st_things: Add example to show how to message handling.

### PR DESCRIPTION
**Pull request by Hyunjun85**
This example has show how to treat request and response.

### CODE REVIEW

File: apps/examples/st_things/msghandling/sample_msghandling_things.c [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include <signal.h>
+#include <st_things/st_things.h>
+#include <iotbus/iotbus_gpio.h>
+
+#define TAG		"LIGHT_THINGS"
+
```
Comment (juitem) [inline apps/examples/st_things/msghandling/sample_msghandling_things.c]: remove spaces & change light_things..

File: apps/examples/st_things/msghandling/sample_msghandling_things.c [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+static const char *g_res_switch = "/capability/switch/main/0";
+
+static const char* g_res_color_temperature_res_uri = "/capability/colorTemperature/main/0";
+static const char* g_res_door_controluri = "/capability/doorControl/main/0";
+static const char* g_res_airquality_res_uri = "/capability/airQualitySensor/main/0";
+static const char* g_res_tone_uri = "/capability/tone/main/0";
+static const char* RES_WWSTFILTERSTATERESURI = "/capability/filterStatus/main/0";
+
```
Comment (juitem) [inline apps/examples/st_things/msghandling/sample_msghandling_things.c]: RES_WWSTFILTERSTATERESURI
aligne naming rule with above variables,

File: apps/examples/st_things/msghandling/sample_msghandling_things.c [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+extern bool handle_set_request_on_resource_doorcontrol(st_things_set_request_message_s* req_msg, st_things_representation_s* resp_rep);
+extern bool handle_get_request_on_resource_wwstairqualitysensorlevelresuri(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep);
+extern bool handle_get_request_on_resource_tone(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep);
+extern bool handle_set_request_on_resource_tone(st_things_set_request_message_s* req_msg, st_things_representation_s* resp_rep);
+
+extern bool handle_get_request_on_resource_wwstfilterstateresuri(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep);
+extern bool handle_set_request_on_resource_wwstfilterstateresuri(st_things_set_request_message_s* req_msg, st_things_representation_s* resp_rep);
+
+#ifdef CONFIG_RESET_BUTTON
+static bool check_reset_button_pin_number(void)
```
Comment (juitem) [inline apps/examples/st_things/msghandling/sample_msghandling_things.c]: CONFIG_RESET_BUTTON, "reset" can be commonly used on each module,
would you please consider to provide different feauture name?

File: apps/examples/st_things/msghandling/sample_msghandling_things.c [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+	st_things_register_request_cb(handle_get_request, handle_set_request);
+	st_things_register_reset_cb(handle_reset_request, handle_reset_result);
+	st_things_register_user_confirm_cb(handle_ownership_transfer_request);
+	st_things_register_things_status_change_cb(handle_things_status_change);
+
+	if (st_things_start() == ST_THINGS_ERROR_NONE) {
+		printf("[%s]=====================================================\n", TAG);
+		printf("[%s]					Stack Started					\n", TAG);
+		printf("[%s]=====================================================\n", TAG);
+	}
```
Comment (juitem) [inline apps/examples/st_things/msghandling/sample_msghandling_things.c]: align decorations

**Comment by Hyunjun85** [comment]
Adds patch.

**Comment by seinfra** [comment]
Target : [b5c8d2e3c33303c80b7d135432b8db6609f7c253] - Code Rule Check Result:
apps/examples/st_things/msghandling/sample_msghandling_user.c:31: ERROR: [IDT_M_TAB] please, no space before tabs

**Comment by seinfra** [comment]
Target : [b5c8d2e3c33303c80b7d135432b8db6609f7c253] - Code Rule Check (C++) OK.

File: apps/examples/st_things/Kconfig [hunk 5]
```diff
@@ -44,11 +44,17 @@ config EXAMPLES_ST_THINGS_BLINK
 	---help---
 		This is a Blink Sample.
 
+config EXAMPLES_ST_THINGS_MSG_HANDLING
+	bool "Message Handling Example"
+	select IOTBUS_GPIO
+	---help---
+		This is a Blink Sample.
```
Comment (JeongJunSik) [inline apps/examples/st_things/Kconfig]: Please change this descript

File: apps/examples/st_things/msghandling/deivce_def.json [hunk 6]
```diff
@@ -0,0 +1,203 @@
+{
+  "device": [
+    {
+      "specification": {
+        "device": {
+          "deviceType": "x.wwst.d.basic",
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/deivce_def.json]: please check this device type

File: apps/examples/st_things/msghandling/sample_msghandling_things.c [hunk 7]
```diff
@@ -0,0 +1,159 @@
+/* ****************************************************************
+ *
+ * Copyright 2017 Samsung Electronics All Rights Reserved.
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_things.c]: 2017 -> 2018

File: apps/examples/st_things/Kconfig (hunk 5 above)
Comment (JeongJunSik) [reply apps/examples/st_things/Kconfig]: If you want to use RESET funtion here, you need to set depends on xxxx at RESET_BUTTON

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 8]
```diff
@@ -0,0 +1,184 @@
+/* ****************************************************************
+*
+* Copyright 2017 Samsung Electronics All Rights Reserved.
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: 2017 -> 2018

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+#include <string.h>
+#include <stdbool.h>
+#include <signal.h>
+#include <st_things/st_things.h>
+
+#define TAG	"MSG_HANDLING_USER"
+
+#define MAX_STR_LEN                                 64
+
+#define PROPERTY_VALUE_SWITCH 				        "power"
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: please check blank align

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+
+#define PROP_DOORSTATE                              "doorState"
+#define PROP_AIRQUALITY                             "airQuality"
+#define PROP_CT                                     "ct"
+#define PROP_RANGE                                   "range"
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: please check blank align

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+#define PROP_TONE                                   "toneValue"
+#define PROP_MACHINESTATES                          "machineStates"
+#define PROP_CURRENTMACHINESTATE                    "currentMachineState"
+
+static char *g_power_status[]                       = { "on", "off" };
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: please check coding rule between { and "

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 12]
```diff
@@ -0,0 +1,184 @@
...
+{
+	if (req_msg->has_property_key(req_msg, PROPERTY_VALUE_SWITCH)) {
+		resp_rep->set_str_value(resp_rep, PROPERTY_VALUE_SWITCH, g_switch_value ? g_power_status[0] : g_power_status[1]);
+	}
+	return true;
+}
+
+bool handle_set_request_on_switch(st_things_set_request_message_s *req_msg, st_things_representation_s *resp_rep)
+{
+
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: please remove this blank line

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 13]
```diff
@@ -0,0 +1,184 @@
...
+			g_switch_value = true;
+		}
+		resp_rep->set_str_value(resp_rep, PROPERTY_VALUE_SWITCH, g_switch_value ? g_power_status[0] : g_power_status[1]);
+		st_things_notify_observers(req_msg->resource_uri);
+	}
+
+	return true;
+}
+
+bool handle_get_request_on_resource_doorcontrol(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep)
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: please check coding rule of * posistion.
st_things_get_request_message_s *req_msg

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 14]
```diff
@@ -0,0 +1,184 @@
...
+{
+	if (req_msg->has_property_key(req_msg, PROP_DOORSTATE)) {
+		resp_rep->set_str_value(resp_rep, PROP_DOORSTATE, g_door_state); // can be one of ["unknown", "open", "closing", "closed", "opening"]
+	}
+	return true;
+}
+
+bool handle_set_request_on_resource_doorcontrol(st_things_set_request_message_s* req_msg, st_things_representation_s* resp_rep)
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: please check coding rule

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 15]
```diff
@@ -0,0 +1,184 @@
...
+
+		resp_rep->set_str_value(resp_rep, PROP_DOORSTATE, g_door_state);
+		st_things_notify_observers(req_msg->resource_uri);
+		free(doorstate);
+	}
+
+	return true;
+}
+
+bool handle_get_request_on_resource_tone(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep)
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: please check coding rule

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 16]
```diff
@@ -0,0 +1,184 @@
...
+{
+	if (req_msg->has_property_key(req_msg, PROP_TONE)) {
+		resp_rep->set_bool_value(resp_rep, PROP_TONE, g_tone);
+	}
+
+	return true;
+}
+
+bool handle_set_request_on_resource_tone(st_things_set_request_message_s* req_msg, st_things_representation_s* resp_rep)
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: please check coding rule

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 17]
```diff
@@ -0,0 +1,184 @@
...
+	if (req_msg->rep->get_bool_value(req_msg->rep, PROP_TONE, &tone)) {
+		g_tone = tone;
+		resp_rep->set_bool_value(resp_rep, PROP_TONE, tone);
+		st_things_notify_observers(req_msg->resource_uri);
+	}
+
+	return true;
+}
+
+bool handle_get_request_on_resource_wwstairqualitysensorlevelresuri(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep)
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: please check coding rule

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 18]
```diff
@@ -0,0 +1,184 @@
...
+		resp_rep->set_double_value(resp_rep, PROP_AIRQUALITY, g_airqulity);
+	}
+	if (req_msg->has_property_key(req_msg, PROP_RANGE)) {
+		resp_rep->set_double_array_value(resp_rep, PROP_RANGE, g_airqulity_range, 2);
+	}
+	return true;
+}
+
+
+bool handle_get_request_on_resource_wwstcolortemperatureresuri(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep)
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: please check coding rule

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 19]
```diff
@@ -0,0 +1,184 @@
...
+	if (req_msg->has_property_key(req_msg, PROP_CT)) {
+		resp_rep->set_int_value(resp_rep, PROP_CT, g_color_temp);
+	}
+	if (req_msg->has_property_key(req_msg, PROP_RANGE)) {
+		resp_rep->set_int_array_value(resp_rep, PROP_RANGE, g_color_temp_range, 2);
+	}
+	return true;
+}
+
+bool handle_set_request_on_resource_wwstcolortemperatureresuri(st_things_set_request_message_s* req_msg, st_things_representation_s* resp_rep)
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: please check coding rule

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 20]
```diff
@@ -0,0 +1,184 @@
...
+{
+	int64_t color_temp = 0;
+	if (req_msg->rep->get_int_value(req_msg->rep, PROP_CT, &color_temp)) {
+		g_color_temp = color_temp;
+		resp_rep->set_int_value(resp_rep, PROP_CT, g_color_temp);
+	}
+	return true;
+}
+
+bool handle_get_request_on_resource_wwstfilterstateresuri(st_things_get_request_message_s* req_msg, st_things_representation_s* resp_rep)
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: please check coding rule

File: apps/examples/st_things/msghandling/sample_msghandling_user.c [hunk 21]
```diff
@@ -0,0 +1,184 @@
...
+		machine_state_list[0] = (char *)"RUNNING";
+		machine_state_list[1] = (char *)"STOP";
+		machine_state_list[2] = (char *)"READY";
+		resp_rep->set_str_array_value(resp_rep, PROP_CURRENTMACHINESTATE, (const char **)machine_state_list, 3);
+		free(machine_state_list);
+	}
+	return true;
+}
+
+bool handle_set_request_on_resource_wwstfilterstateresuri(st_things_set_request_message_s* req_msg, st_things_representation_s* resp_rep)
```
Comment (JeongJunSik) [inline apps/examples/st_things/msghandling/sample_msghandling_user.c]: please check coding rule

**Comment by seinfra** [comment]
Target : [48b540d506f25ab027ffa7f97c20525e7cfaebd6] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [48b540d506f25ab027ffa7f97c20525e7cfaebd6] - Code Rule Check (C++) OK.

**Pull request merged**